### PR TITLE
Update readme file

### PR DESCRIPTION
**Pull request by abdahmed22**
### Description

Update client readme file to include introduction, features, setup and tests. 
### Changes

- Add logo
- Add badges
- Introduction describing the project 
- Prerequisites
- How to Setup on a local machine
- How to run tests

### Related Issues

- https://github.com/codescalersinternships/test-tracker/issues/36

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -1,81 +1,210 @@
-# Vuetify (Default)
+# Test Tracker
 
-This is the official scaffolding tool for Vuetify, designed to give you a head start in building your new Vuetify application. It sets up a base template with all the necessary configurations and standard directory structure, enabling you to begin development without the hassle of setting up the project from scratch.
+## Introduction
 
-## ❗️ Important Links
+TestTracker allows the entire team to manage their test plans, requirements, test cases and test runs in a simple and efficient way. All essential tools to manage your test cases all in one place.
```
Comment (maayarosama) [inline client/README.md:5]:
```suggestion
TestTracker allows the QA team to manage their test plans, requirements, test cases and test runs in a simple and efficient way. All essential tools to manage your test cases all in one place. 
```

File: client/README.md [hunk 2]
```diff
@@ -1,81 +1,210 @@
...
 
-## ❗️ Important Links
+TestTracker allows the entire team to manage their test plans, requirements, test cases and test runs in a simple and efficient way. All essential tools to manage your test cases all in one place. 
 
-- 📄 [Docs](https://vuetifyjs.com/)
-- 🚨 [Issues](https://issues.vuetifyjs.com/)
-- 🏬 [Store](https://store.vuetifyjs.com/)
-- 🎮 [Playground](https://play.vuetifyjs.com/)
-- 💬 [Discord](https://community.vuetifyjs.com)
+## Features
```
Comment (maayarosama) [inline client/README.md:7]: Th features part isn't needed, you can remove it

File: client/README.md [hunk 3]
```diff
@@ -1,81 +1,210 @@
...
+        DATABASE_PASSWORD=
+        DATABASE_HOST=
+        DATABASE_PORT=
+        
+        DJANGO_SUPERUSER_EMAIL=
+        DJANGO_SUPERUSER_PASSWORD=
+  
+4. Run the following command to install dependencies:
+      ```console
+   user@user-VirtualBox:~$ make install
```
Comment (maayarosama) [inline client/README.md:178]:
```suggestion
  ```make install```
```
apply to all other commands

File: client/README.md
Comment (maayarosama) [inline client/README.md:1]: add prerequisites section and move the readme file to the root directory

**Review by maayarosama — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1,25 +1,137 @@
...
+
+---
+<p align="center">
+  <img src="https://img.shields.io/badge/-TypeScript-black?style=for-the-badge&logoColor=white&logo=typescript&color=2F73BF" alt="Typescript"/>
+  <img src="https://img.shields.io/badge/-Vue.js-4fc08d?style=for-the-badge&logo=vuedotjs&logoColor=white" alt="Vue"/>
+  <img src="https://img.shields.io/badge/Django-092E20?style=for-the-badge&logo=django&logoColor=green" alt="Django"/>
+  <img src="https://img.shields.io/badge/postgresql-4169e1?style=for-the-badge&logo=postgresql&logoColor=white" alt="Postgresql"/>
+</p>
+
+## <a name="table-contents">Table of Contents</a>
```
Comment (zaelgohary) [inline README.md:16]:
```suggestion
## Table of Contents
```

Comment (zaelgohary) [reply README.md:16]: This applies on all headers.

File: README.md [hunk 5]
```diff
@@ -1,25 +1,137 @@
...
+
+1.  [Introduction](#introduction)
+2.  [Prerequisites](#prerequisites)
+3.  [Setup on your local machine](#local-setup)
+4.  [Setup using docker](#docker-setup)
+5.  [Tests](#tests)
```
Comment (zaelgohary) [inline README.md:22]:
```suggestion
1. [Introduction](#introduction)
2. [Prerequisites](#prerequisites)
3. [Setup on Your Local Machine](#setup-on-your-local-machine)
4. [Setup Using Docker](#setup-using-docker)
5. [Tests](#tests)
```

File: README.md [hunk 6]
```diff
@@ -1,25 +1,137 @@
...
+TestTracker allows the Quality Assurance team to manage their test plans, requirements, test cases and test runs in a simple and efficient way. All essential tools to manage your test cases all in one place. 
+
+## <a name="prerequisites">Prerequisites</a>
+
+Make sure you have installed all of the following prerequisites on your machine:
+* Git - [Download & Install Git](https://git-scm.com/downloads). OSX and Linux machines typically have this already installed.
+* Node.js - [Download & Install Node.js](https://nodejs.org/en/download/) and the pnpm package manager. If you encounter any problems, you can also use this [GitHub Gist](https://gist.github.com/isaacs/579814) to install Node.js.
+* Python3 - [Download & Install Python3](https://www.python.org/downloads/).
+* Poetry - [Download & Install Poetry](https://python-poetry.org/docs/).
+* Docker - [Download & Install Docker](https://docs.docker.com/engine/install/). Skip this step if you plan to use your local machine to run Test Tracker.
```
Comment (zaelgohary) [inline README.md:36]:
```suggestion
- **Git** - [Download & Install Git](https://git-scm.com/downloads). OSX and Linux machines typically have this already installed.
- **Node.js** - [Download & Install Node.js](https://nodejs.org/en/download/) and the pnpm package manager. If you encounter any problems, you can also use this [GitHub Gist](https://gist.github.com/isaacs/579814) to install Node.js.
- **Python3** - [Download & Install Python3](https://www.python.org/downloads/).
- **Poetry** - [Download & Install Poetry](https://python-poetry.org/docs/).
- **Docker** - [Download & Install Docker](https://docs.docker.com/engine/install/). Skip this step if you plan to use your local machine to run Test Tracker.

```

File: README.md [hunk 7]
```diff
@@ -1,25 +1,137 @@
...
+
+
+## <a name="local-setup">Setup on your local machine</a>
```
Comment (zaelgohary) [inline README.md:39]:
```suggestion
## Setup on Your Local Machine
```

File: README.md [hunk 8]
```diff
@@ -1,25 +1,137 @@
...
+
+1. Clone the Repository to a directory of your choice.
+2. Create .env file using the following template:
```
Comment (zaelgohary) [inline README.md:42]:
```suggestion
1. Clone the repository to a directory of your choice.
2. Create a `.env` file using the following template:
```

File: README.md [hunk 9]
```diff
@@ -1,25 +1,137 @@
...
+        
+        # === Database ===
+        DATABASE_NAME=
+        DATABASE_USER=
+        DATABASE_PASSWORD=
+        DATABASE_HOST=
+        DATABASE_PORT=
+        
+        DJANGO_SUPERUSER_EMAIL=
+        DJANGO_SUPERUSER_PASSWORD=
```
Comment (zaelgohary) [inline README.md:66]:
```suggestion
    ```env
    # === General ===        
    DJANGO_SECRET_KEY=
    SERVER_DOMAIN_NAME=
    SERVER_DOMAIN_NAME_API=
    CLIENT_DOMAIN_NAME=
    ENV=production
    DJANGO_DEBUG=OFF
    
    # === Email ===
    EMAIL=
    EMAIL_PASSWORD=
    EMAIL_HOST=
    
    # === Database ===
    DATABASE_NAME=
    DATABASE_USER=
    DATABASE_PASSWORD=
    DATABASE_HOST=
    DATABASE_PORT=
    
    DJANGO_SUPERUSER_EMAIL=
    DJANGO_SUPERUSER_PASSWORD=
    ```
```

File: README.md [hunk 10]
```diff
@@ -1,25 +1,137 @@
...
+  
+4. Run the following command to install dependencies:
+      console
+    make install
```
Comment (zaelgohary) [inline README.md:70]:
```suggestion
    ```console
    make install
    ```
```

File: README.md [hunk 11]
```diff
@@ -1,25 +1,137 @@
...
+    make user
+   
+7. Run the following command to run server:
+      console
+    make runserver
+   
+8. Open a new terminal tab.
+9. Run the following command to run client:
+      console
+    make runclient
```
Comment (zaelgohary) [inline README.md:87]: Same here in all console commands.

**Review by zaelgohary — CHANGES_REQUESTED** [review]

File: client/README.md [hunk 12]
```diff
@@ -1,81 +1,210 @@
...
+  Search for a project
+
+  <p>&nbsp;</p>
+</details>
+
+<details>
+  <summary>Test Plans</summary>
+  <p>&nbsp;</p>
+
+  Create a defaullt test plan
```
Comment (zaelgohary) [inline client/README.md:74]:
```suggestion
  Create a default test plan
```

File: client/README.md [hunk 13]
```diff
@@ -1,81 +1,210 @@
...
-- 🚦 **Routing and Layouts**: Utilizes Vue Router for SPA navigation and vite-plugin-vue-layouts for organizing Vue file layouts. [Vue Router](https://router.vuejs.org/) | [vite-plugin-vue-layouts](https://github.com/JohnCampionJr/vite-plugin-vue-layouts)
-- 💻 **Enhanced Development Experience**: Benefit from TypeScript's static type checking and the ESLint plugin suite for Vue, ensuring code quality and consistency. [TypeScript](https://www.typescriptlang.org/) | [ESLint Plugin Vue](https://eslint.vuejs.org/)
-- ⚡ **Next-Gen Tooling**: Powered by Vite, experience fast cold starts and instant HMR (Hot Module Replacement). [Vite](https://vitejs.dev/)
-- 🧩 **Automated Component Importing**: Streamline your workflow with unplugin-vue-components, automatically importing components as you use them. [unplugin-vue-components](https://github.com/antfu/unplugin-vue-components)
-- 🛠️ **Strongly-Typed Vue**: Use vue-tsc for type-checking your Vue components, and enjoy a robust development experience. [vue-tsc](https://github.com/johnsoncodehk/volar/tree/master/packages/vue-tsc)
+  Logout
```
Comment (zaelgohary) [inline client/README.md:21]: Convert to an unordered list.

File: client/README.md [hunk 14]
```diff
@@ -1,81 +1,210 @@
...
 
-### Starting the Development Server
+  Delete a member
 
-To start the development server with hot-reload, run the following command. The server will be accessible at [http://localhost:3000](http://localhost:3000):
 
-```bash
-yarn dev
-```
+  Search for a member
```
Comment (zaelgohary) [inline client/README.md:36]: Same here.

File: client/README.md [hunk 15]
```diff
@@ -1,81 +1,210 @@
...
-- [Becoming a subscriber on Tidelift](https://tidelift.com/subscription/npm/vuetify)
-- [Making a one-time donation with Paypal](https://paypal.me/vuetify)
+  Add new test run
 
-## 📑 License
-[MIT](http://opensource.org/licenses/MIT)
+  Delete a project
 
-Copyright (c) 2016-present Vuetify, LLC
+  Search for a project
```
Comment (zaelgohary) [inline client/README.md:65]: and here.

File: client/README.md [hunk 16]
```diff
@@ -1,81 +1,210 @@
...
-```bash
-yarn build
-```
+  View Test runs
 
-(Repeat for npm, pnpm, and bun with respective commands.)
+  View recent activities
 
-Once the build process is completed, your application will be ready for deployment in a production environment.
+  Add new test plan
```
Comment (zaelgohary) [inline client/README.md:55]:
```suggestion
  Add a new test plan
```

File: client/README.md [hunk 17]
```diff
@@ -1,81 +1,210 @@
...
 
-## 💪 Support Vuetify Development
+  Add new test requirments
```
Comment (zaelgohary) [inline client/README.md:57]:
```suggestion
  Add new test requirements
```

File: client/README.md [hunk 18]
```diff
@@ -1,81 +1,210 @@
...
 
-This project is built with [Vuetify](https://vuetifyjs.com/en/), a UI Library with a comprehensive collection of Vue components. Vuetify is an MIT licensed Open Source project that has been made possible due to the generous contributions by our [sponsors and backers](https://vuetifyjs.com/introduction/sponsors-and-backers/). If you are interested in supporting this project, please consider:
+  Add new test suite
```
Comment (zaelgohary) [inline client/README.md:59]:
```suggestion
  Add a new test suite
```

File: client/README.md [hunk 19]
```diff
@@ -1,81 +1,210 @@
...
 
-- [Requesting Enterprise Support](https://support.vuetifyjs.com/)
-- [Sponsoring John on Github](https://github.com/users/johnleider/sponsorship)
-- [Sponsoring Kael on Github](https://github.com/users/kaelwd/sponsorship)
-- [Supporting the team on Open Collective](https://opencollective.com/vuetify)
-- [Becoming a sponsor on Patreon](https://www.patreon.com/vuetify)
-- [Becoming a subscriber on Tidelift](https://tidelift.com/subscription/npm/vuetify)
-- [Making a one-time donation with Paypal](https://paypal.me/vuetify)
+  Add new test run
```
Comment (zaelgohary) [inline client/README.md:61]:
```suggestion
  Add a new test run
```

File: client/README.md [hunk 20]
```diff
@@ -1,81 +1,210 @@
...
-(Repeat for npm, pnpm, and bun with respective commands.)
+<details>
+  <summary>Projects</summary>
+  <p>&nbsp;</p>
 
-> Add NODE_OPTIONS='--no-warnings' to suppress the JSON import warnings that happen as part of the Vuetify import mapping. If you are on Node [v21.3.0](https://nodejs.org/en/blog/release/v21.3.0) or higher, you can change this to NODE_OPTIONS='--disable-warning=5401'. If you don't mind the warning, you can remove this from your package.json dev script.
+  Create a project
 
-### Building for Production
+  View Statistics
```
Comment (zaelgohary) [inline client/README.md:47]:
```suggestion
  View statistics
```

File: client/README.md [hunk 21]
```diff
@@ -1,81 +1,210 @@
...
+
+  Create a defaullt test plan
+
+  Create a custom test plan
+
+  Update a test plan
+
+  Delete a test plan
+
+  Search for a test plan
```
Comment (zaelgohary) [inline client/README.md:82]: convert to an unordered list.

File: client/README.md (hunk 12 above)
Comment (zaelgohary) [inline client/README.md:74]:
```suggestion
  Create a default test plan
```

File: client/README.md [hunk 22]
```diff
@@ -1,81 +1,210 @@
...
+
+  Delete a requirment document
+
+  Search for a requirment document
+  
+  Add a new requirment
+
+  Delete a requirment
+
+  Search for a requirment
```
Comment (zaelgohary) [inline client/README.md:101]: Same here.

File: client/README.md [hunk 23]
```diff
@@ -1,81 +1,210 @@
...
+  Delete a test plan
+
+  Search for a test plan
+  
+  <p>&nbsp;</p>
+</details>
+
+<details>
+  <summary>Requirment Documents</summary>
+  <p>&nbsp;</p>
```
Comment (zaelgohary) [inline client/README.md:89]:
```suggestion
```

File: client/README.md [hunk 24]
```diff
@@ -1,81 +1,210 @@
...
+
+  Create a custom test plan
+
+  Update a test plan
+
+  Delete a test plan
+
+  Search for a test plan
+  
+  <p>&nbsp;</p>
```
Comment (zaelgohary) [inline client/README.md:84]:
```suggestion

```

File: client/README.md [hunk 25]
```diff
@@ -1,81 +1,210 @@
...
+
+  Search for a requirment document
+  
+  Add a new requirment
+
+  Delete a requirment
+
+  Search for a requirment
+  
+  <p>&nbsp;</p>
```
Comment (zaelgohary) [inline client/README.md:103]:
```suggestion
```

File: client/README.md [hunk 26]
```diff
@@ -1,81 +1,210 @@
...
+</details>
+
+<details>
+  <summary>Test Suites</summary>
+  <p>&nbsp;</p>
```
Comment (zaelgohary) [inline client/README.md:108]:
```suggestion
```

File: client/README.md [hunk 27]
```diff
@@ -1,81 +1,210 @@
...
+
+  Assign to a test plan
+
+  Add test case section
+
+  Add new test case 
+
+  Add existing test case
+
+  Search for a test case
```
Comment (zaelgohary) [inline client/README.md:124]: same here.

File: client/README.md [hunk 28]
```diff
@@ -1,81 +1,210 @@
...
+  
+  <p>&nbsp;</p>
```
Comment (zaelgohary) [inline client/README.md:126]:
```suggestion
```

File: client/README.md [hunk 29]
```diff
@@ -1,81 +1,210 @@
...
+</details>
+
+<details>
+  <summary>Test runs</summary>
+  <p>&nbsp;</p>
```
Comment (zaelgohary) [inline client/README.md:131]:
```suggestion
```

File: client/README.md [hunk 30]
```diff
@@ -1,81 +1,210 @@
...
+
+  Create a test run
+
+  Delete a test run
+
+  Search for a test run
+  
+  View test run report
+
+  Involve a team member
```
Comment (zaelgohary) [inline client/README.md:141]: same.

File: client/README.md [hunk 31]
```diff
@@ -1,81 +1,210 @@
...
+  
+  <p>&nbsp;</p>
```
Comment (zaelgohary) [inline client/README.md:143]:
```suggestion
```

File: client/README.md [hunk 32]
```diff
@@ -1,81 +1,210 @@
...
+
+  Search for a test suite
+
+  Assign to a test plan
+
+  Add test case section
+
+  Add new test case 
+
+  Add existing test case
```
Comment (zaelgohary) [inline client/README.md:122]:
```suggestion
- Add a test case section
- Add a new test case
- Add an existing test case
```

File: client/README.md (hunk 12 above)
Comment (abdahmed22) [reply client/README.md:74]: I removed the features section

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**